lib/galaxy_utils/sequence/fastq.py

```diff
@@ -673,13 +677,14 @@ def join( self, read1, read2 ):
         if rval.sequence_space == 'color':
             #need to handle color space joining differently
             #convert to nuc space, join, then convert back
-            rval.sequence = rval.convert_base_to_color_space( new_read1.convert_color_to_base_space( new_read1.sequence ) + new_read2.convert_color_to_base_space( new_read2.sequence ) )
+            rval.sequence = rval.convert_base_to_color_space( new_read1.convert_color_to_base_space( new_read1.sequence ) + self.paste_sequence + new_read2.convert_color_to_base_space( new_read2.sequence ) )
         else:
-            rval.sequence = new_read1.sequence + new_read2.sequence
+            rval.sequence = new_read1.sequence + self.paste_sequence + new_read2.sequence
         if force_quality_encoding == 'ascii':
-            rval.quality = new_read1.quality + new_read2.quality
+            rval.quality = new_read1.quality + self.paste_ascii_quality + new_read2.quality
         else:
-            rval.quality = "%s %s" % ( new_read1.quality.strip(), new_read2.quality.strip() )
+            rval.quality = "%s %s" % ( new_read1.quality.strip(), self.paste_decimal_quality )
+            rval.quality = ( "%s %s" % ( rval.quality.strip(), new_read2.quality.strip() ) ).strip()
```

Comment:
If ``rval`` has no quality, then the space in ``"%s %s"`` becomes a leading space.

If ``new_read2`` has no quality, then the space in ``"%s %s"`` becomes a trailing space.

So the final strip looks like it could be useful?